### PR TITLE
Simple "string" representation for number literals

### PR DESCRIPTION
**Pull request by paluh**
As we don't need to perform any calculations on typelevel numbers I think that we can represent number literals directly in `Symbols`. What do you think?

### CODE REVIEW

**Comment by paluh** [comment]
I'm not sure if `Null` and `Undefined` should be included here...

**Comment by jvliwanag** [comment]
Awesome! Yup, I imagined number literals written as symbols as well with format checking should be ok.

I was hoping other more commonly used packages would have `Null` and `Undefined` being their own types already defined -- but that doesn't seem to be the case. So best to add those here as well.

File: src/Literals/Boolean.purs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+module Literals.Boolean where
```
Comment (jvliwanag) [inline src/Literals/Boolean.purs]: Let's add export list. Also, add `toBoolean`

File: src/Literals/Null.purs [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module Literals.Null where
```
Comment (jvliwanag) [inline src/Literals/Null.purs:1]: Add exports

**Comment by jvliwanag** [comment]
I'm thinking if we should name this as `NumberLiteral`, same goes with the others. That way, there's no need to qualify imports. We can also make a `Literals` top level module that imports from the other modules.

Using this package, ideally, no one should write `Literal "2.0"` anyway in their code. It should instead read as `Number.Literal "2.0"` or `NumberLiteral "2.0"` if we choose to rename it.

Alternatively, `StringLit`/`NumberLit` or `SLiteral`/`NLiteral` or `SLit`/`NLit`? Chances are users of this package will be ffi bindings, and together with OneOf. Thus ending up with something like: 

`StringLiteral "left" |+| StringLiteral "right" |+| StringLiteral "top" |+| StringLiteral "bottom"`

might just be too wordy. I'm ok exploring this in a diff issue though. Or just having type aliases.

**Comment by paluh** [comment]
> I'm thinking if we should name this as NumberLiteral, same goes with the others. That way, there's no need to qualify imports. We can also make a Literals top level module that imports from the other modules.

My coding style is usually other way around - I mean I'm using module namespace as a way to disambiguate entities so I can reuse names between modules but it is again just personal preference :-P

There is also `StrLit` and `NumLit`. Which one should I finally use? :-)

File: src/Literals/Boolean.purs (hunk 1 above)
Comment (paluh) [reply src/Literals/Boolean.purs]: I'm not sure about `toBoolean`. Should I also introduce `BooleanLiteral` here like:
```purescript
foreign import kind Boolean
foreign import data True ∷ Boolean
foreign import data False ∷ Boolean

foreign import BooleanLiteral :: Boolean -> Type
```
plus something like:

```purescrpit
toBoolean :: forall a. BooleanLiteral a -> Boolean
```

Have I missed the point? ;-)

**Comment by jvliwanag** [comment]
how about let's do:

```purescript
foreign import data Literal :: Type -> Symbol -> Type

type StringLit sym = Literal String sym
type NumberLit sym = Literal Number sym
type BooleanLit sym = Literal Boolean sym

toValue :: forall typ sym. Literal typ sym -> typ
toValue = unsafeCoerce
```

**Comment by paluh** [comment]
Yeah, I think that is the way to go :-)

**Comment by paluh** [comment]
As usual - sorry for this delay ;-) I hope I've incorporated all your suggestions.

**Comment by jvliwanag** [comment]
Awesome. Thanks!

**Comment by paluh** [comment]
I think I've forgotten to add `Null` and `Undefined` exports to the main `Literals` module. Do you think I should provide consistent encoding for them too? I mean by using something like `Literal Undefined "undefined"` and `Literal Null "null"`?

**Comment by jvliwanag** [comment]
Probably better to include `Null` and `Undefined` onto the main `Literals` module.

I don't think there's a need for `Literal Undefined "undefined"`, since that would simply just point back to the `Undefined` type.
